### ISSUE TITLE
Player doesn't refresh file metadata after it was updated by plugin

### ISSUE THREAD

**Issue by Adonai** [vanilla-music/vanilla#879]
So, okay, Tag Editor plugin calls this after it changes a file:
```
MediaScannerConnection.scanFile(mContext, new String[]{persisted.getAbsolutePath()}, null, null);
```

Why isn't library and playback activity refreshed after that? I have to wait ~1-2 days before I actually see the changes I made to the file.

**Comment by adrian-bl** [comment]
The media library subscribes to updates via
https://github.com/vanilla-music/vanilla/blob/master/app/src/main/java/ch/blinkenlights/android/medialibrary/MediaScanner.java#L108

We then check whether files got deleted or not (https://github.com/vanilla-music/vanilla/blob/master/app/src/main/java/ch/blinkenlights/android/medialibrary/MediaScanner.java#L359)

If no files were deleted, we check for recent modifications:
https://github.com/vanilla-music/vanilla/blob/master/app/src/main/java/ch/blinkenlights/android/medialibrary/MediaScanner.java#L379

So in theory that should work? (I suppose the plugin editor does not try to preserve the mtime? Or maybe android fails to update the DB for in-place modifications?).

But looks like implementing the newer onChange interface (https://developer.android.com/reference/android/database/ContentObserver.html#onChange(boolean,%20android.net.Uri)) would also be a good idea regardless.

**Comment by Adonai** [comment]
OK, so what I see is that we're totally lacking updates to the timeline in addition to database and library adapter updates. Will work on correcting that.

**Comment by Adonai** [comment]
https://github.com/vanilla-music/vanilla/blob/master/app/src/main/java/ch/blinkenlights/android/medialibrary/MediaScanner.java#L108

BTW here `descendants` parameter should be `true` or quickscans are not getting scheduled

**Comment by adrian-bl** [comment]
I don't think that the value matters in this case as we observe a 'virtual' uri:
I just tested it and `onChange()` does get called.

**Comment by adrian-bl** [comment]
So i just tested this with the tag editor plugin on my Android 4.4 test device and the changes made by the plugin are picked up after ~1 sec

Log with some additional info:
```
V/VanillaMusic( 2984): onChange happened!   // observer kicked in
V/VanillaMusic( 2984): lastSeenDbSize = 109  // lastSeen == current, so we call RPC_NATIVE_VRFY
V/VanillaMusic( 2984): currentDbSize  = 109
V/VanillaMusic( 2984): xxx --- starting scan of type 104

V/VanillaMusic( 2984): Extracting tags from /storage/emulated/0/Music/katamari/katamari_tribute_ps3/203 Everlasting Love + You.mp3
V/VanillaMusic( 2984): MediaScanner: inserted /storage/emulated/0/Music/katamari/katamari_tribute_ps3/203 Everlasting Love + You.mp3
V/VanillaMusic( 2984): rpcNativeVerify with cursor = android.content.ContentResolver$CursorWrapperInner@4212c398
V/VanillaMusic( 2984): NativeLibraryScanner finished, mtime mark is now at 1543860007
```

**Comment by Adonai** [comment]
Just tested it on Oreo emulator and it doesn't get called with `notifyForDescendants` set to `false` and a file inside "Music" folder

**Comment by adrian-bl** [comment]
Interesting! So does it work if you set it to true?

**Comment by adrian-bl** [comment]
Tested this on api28 and setting notifyForDescendandts = true is indeed required.

e997ac9207b560484d3fdbd71a0d38864c13cd99 fixes this - good catch. thanks!

**Comment by Adonai** [comment]
Yes! I'm preparing a PR right now

**Comment by Adonai** [comment]
Adrian, need your eyes on that linked PR

**Comment by adrian-bl** [comment]
I'll try to have a look at it this weekend
